packages/core/src/modules/oob/OutOfBandApi.ts

```diff
@@ -76,6 +76,15 @@ export interface ReceiveOutOfBandInvitationConfig {
   reuseConnection?: boolean
   routing?: Routing
   acceptInvitationTimeoutMs?: number
+  isImplicit?: boolean
+}
+
+export type ReceiveOutOfBandInvitationConfig = Omit<BaseReceiveOutOfBandInvitationConfig, 'isImplicit'>
+
+export interface ReceiveOutOfBandImplicitInvitationConfig
+  extends Omit<BaseReceiveOutOfBandInvitationConfig, 'isImplicit' | 'reuseConnection'> {
```

Comment:
Why don't we allow `reuseConnection`? If I'm already connected I may or may not reuse the connection right?

Reply:
I set this before moving to the idea of using `id = did `for all OOBInvitations related to the same did. I think it can be re-added, but it can add some complexity: suppose you have a connection related to a public did and call this method with `reuseConnection = false`, meaning that it will create a new connection related to the same public did. As a result, we will have two connections related to the same did. If you call this method again with `reuseConnection = true`, which of those two connections should it reuse?

I think it´s valuable to have this feature enabled in an automatic way (otherwise it's the consumer app who should check if there is an existing connection before calling this method). Maybe we can support it and, if two or more connections are found, we throw an error?

Reply:
The problem I see here is that `reuseConnection` parameter is related to the reuse messages as specified in [RFC 434](https://github.com/hyperledger/aries-rfcs/blob/main/features/0434-outofband/README.md#reuse-messages) (`handshake-reuse` and `handshake-reuse-accepted`). This one uses OOB Invitation `pthid` as the way of identifying the invitee... but in an implicit invitation, the pthid will be the public did of the inviter :thinking: .

This makes me think that this flow is not possible or intended to work with implicit invitations. Any clarification will be certainly welcome (my brains are as fried as yours nowadays :stuck_out_tongue:)

Reply:
Yes makes sense. I think we may also be able to just return an existing connection based on `reuseConnection`. if you pass `reuseConnection=true` it will just look if we already have a connection with this did, but if you pass false it will always create a new connection. With an OOB invitation you MUST use the handshake-reuse. But this approach is perfect for now